macros/misc/randomNamesPronouns.pl

```diff
@@ -0,0 +1,648 @@
+	[ Tyrese     => 'he' ],
+	[ Valentina  => 'she' ],
+	[ Xzavier    => 'he' ],
+	[ Yair       => 'he' ],
+	[ Yuliana    => 'she' ],
+	[ Zane       => 'he' ],
+	[ Zoey       => 'she' ]
+);
+
+@lastnames = qw(
+	Adams   Allen     Alvarez    Anderson Bailey  Baker    Bennet    Brooks
+	Brown   Campbell  Carter     Castillo Chavez  Clark    Collins   Cook
+	Cooper  Cox       Cruz       Davis    Diaz    Edwards  Evans     Flores
+	Foster  Garcia    Gomez      Gonzales Gray    Green    Gutierrez Hall
+	Harris  Hernandez Hill       Howard   Hughes  Jackson  James     Jimenez
+	Johnson Jones     Kelly      Kim      King    Lee      Lewis     Long
+	Lopez   Martin    Martinez   Mendoza  Miller  Mitchell Moore     Morales
+	Morgan  Morris    Murphy     Myers    Nelson  Nguyen   Ortiz     Parker
+	Patel   Perez     Peterson   Phillips Pina    Price    Ramirez   Ramos
+	Reed    Reyes     Richardson Rivera   Roberts Robinson Rodriguez Rogers
+	Ross    Ruiz      Sanchez    Sanders  Scott   Smith    Stewart   Sullivan
+	Taylor  Thomas    Thompson   Torres   Turner  Walker   Ward      Watson
+	White   Williams  Wilson     Wood     Wright  Young
+);
+
+=head2 randomPerson
+
+Returns a person as a Person object from a list in the macro.
+
+=head3 Examples and options
+
+=over
+
+=item * No arguments returns a single random person
+
+Example
+
+    randomPerson()
+
+=item * C<S<< n => k >>> returns an array of k Person objects with unique names.
+
+Example
+
+    randomPerson(n=>5)
+
+returns an array of 5 Person objects with unique names.
+
+=item * C<S<< names => arrayref >>> returns either a single Person or array of Persons
+from the given names
+
+If the arrayref is in the form of C<S<< [['name1' => 'pronoun1'],['name2' => 'pronoun2'], ...] >>> then
+a person selected randomly from the given list is returned.
+
+Example:
+
+    $a = randomPerson(names => [[ Bart => 'he' ], [ Lisa => 'she' ], [ Matty => 'they' ]]);
+
+Alternatively, each person name/pronoun can be set as an hashref.
+For example the above can be written:
+
+    $a = randomPerson(names => [
+        { name => 'Bart', pronoun => 'he' },
+        { name => 'Lisa', pronoun => 'she' },
+        { name => 'Matty', pronoun => 'they' }
+    ]);
+
+If the pronoun is missing using either arrayrefs or hashrefs, then a pronoun is determined randomly.
+Each of the following are legal
+
+    $p1 = randomPerson( names => ['Larry', 'Moe', 'Curly']);
+    $p2 = randomPerson( names => ['Larry', [ Moe => 'he' ], 'Curly']);
+    $p3 = randomPerson( names => [{ name => 'Larry'}, { name => 'Moe' }, { name => 'Curly' }]);
+    $p4 = randomPerson( names => [{ name => 'Larry'}, { name => 'Moe', pronoun => 'he' }, { name => 'Curly' }]);
+
+And the option C<n> can be used to return an array with that number of persons.
+
+    @p = randomPerson(n => 2, names => [[ Bart => 'he' ], [ Lisa => 'she' ], [ Matty => 'they' ]]);
+
+=back
+
+=cut
+
+sub randomPerson {
+	my %options = (n => 1, @_);
+	my @persons = ();
```

Comment:
Please drop the `= ()` part.  In Perl hashes and arrays are always created initialized to an empty hash or array.  Adding `= ()` does nothing except to add code clutter.

Reply:
yep.